src/main/java/randoop/generation/DemandDrivenInputCreator.java

```diff
@@ -0,0 +1,406 @@
+   */
+  private boolean onlyReceivers;
+
+  /** Constructs a new {@code DemandDrivenInputCreator} object. */
+  public DemandDrivenInputCreator(
+      SequenceCollection sequenceCollection, boolean exactTypeMatch, boolean onlyReceivers) {
+    this.sequenceCollection = sequenceCollection;
+    this.secondarySequenceCollection = new SequenceCollection();
+    this.exactTypeMatch = exactTypeMatch;
+    this.onlyReceivers = onlyReceivers;
+  }
+
+  /**
+   * Performs a demand-driven approach for constructing input objects of a target type, when the
+   * sequence collection contains no objects of that type.
+   *
+   * <p>This method processes all available constructors and methods to identify possible ways to
+   * create objects of the {@code targetType}. For each method or constructor, it attempts to
+   * generate a sequence by searching for necessary inputs from the provided sequence collection,
+   * executing the sequence, and, if successful, storing it in the sequence collection for future
+   * use.
+   *
+   * <p>At the end of the process, it filters and returns the sequences that produce objects of the
+   * {@code targetType}, if any are found.
+   *
+   * <p>Here is the demand-driven algorithm in more detail:
+   *
+   * <ol>
+   *   <li>Initialize a worklist with the {@code targetType} and user-specified classes.
+   *   <li>Process types in the worklist:
+   *       <ul>
+   *         <li>Remove the next type. Skip this type if already processed or if it is a
+   *             non-receiver type.
+   *         <li>Identify constructors and methods that can produce objects of the current type or
+   *             the target type.
+   *         <li>Add input parameter types of these producer methods to the worklist.
+   *       </ul>
+   *   <li>For each producer method, try to find sequences for its inputs in the sequence
+   *       collection.
+   *   <li>If inputs are found, create and execute a sequence. Store successful sequences.
+   *   <li>Return sequences that produce the {@code targetType}.
+   * </ol>
+   *
+   * <p>Note that a single call to this method may not be sufficient to construct the target type,
+   * even when possible sequences exist. The method may need to be called multiple times to
+   * successfully construct the object. If no sequences are found in a single run but the sequence
+   * can possibly be constructed, the call to this method often constructs intermediate sequences
+   * and stores them in the sequence collection that can help future runs of demand-driven input
+   * creation to succeed.
+   *
+   * <p>Invariant: This method is only called when the component sequence collection ({@link
+   * ComponentManager#gralComponents}) lacks a sequence that creates an object of a type compatible
+   * with the one required by the forward generator. See {@link
+   * randoop.generation.ForwardGenerator#selectInputs(TypedOperation)}.
+   *
+   * @param targetType the type of objects to create
```

Comment:
Please state the side effects of this method, which might be global or might be only to the state of the receiver object.